sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala

```diff
@@ -104,9 +107,7 @@ case class ParquetPartitionReaderFactory(
   }
 
   override def supportColumnarReads(partition: InputPartition): Boolean = {
-    sqlConf.parquetVectorizedReaderEnabled && sqlConf.wholeStageEnabled &&
-      resultSchema.length <= sqlConf.wholeStageMaxNumFields &&
-      resultSchema.forall(_.dataType.isInstanceOf[AtomicType])
+    supportsColumnar
```

Comment:
should we just inline it here instead of declaring a private field `supportsColumnar`? since it's only used in this method.

Reply:
Since this is called per partition, I figured it made since to make a private field to calculate `!WholeStageCodegenExec.isTooManyFields(sqlConf, resultSchema)` once instead of every partition, since this doesn't actually depend on the partition

Reply:
I see, make sense.